agent/lib/kontena/launchers/etcd.rb

```diff
@@ -134,12 +143,83 @@ def create_container(image, info)
         }
       )
       container.start
-      Celluloid::Notifications.publish('dns:add', {id: container.id, ip: weave_ip, name: 'etcd.kontena.local'})
+      Celluloid::Notifications.publish('dns:add', {id: container.id, ip: weave_ip(info), name: 'etcd.kontena.local'})
       info "started etcd service"
       @running = true
       container
     end
 
+    # Removes possible previous member with the same IP
+    #
+    # @param [String] node weave ip
+    # @return [String] the state of the cluster member
+    def update_membership(info)
```

Comment:
Absolutely. I'll break out at least the rescue-retry loop to find online etcd node.